lighthouse-core/lib/network-request.js

```diff
@@ -341,10 +363,89 @@ module.exports = class NetworkRequest {
     // Bail if we somehow already have transfer size data.
     if (!global.isLightrider) return;
 
-    const totalFetchedSize = this.responseHeaders.find(item => item.name === 'X-TotalFetchedSize');
+    const totalFetchedSize = this.responseHeaders.find(item => item.name === HEADER_FETCHED_SIZE);
     // Bail if the header was missing.
     if (!totalFetchedSize) return;
-    this.transferSize = parseFloat(totalFetchedSize.value);
+    const floatValue = parseFloat(totalFetchedSize.value);
+    // Bail if the header cannot be parsed
+    if (isNaN(floatValue)) return;
+    this.transferSize = floatValue;
+  }
+
+  /**
+   * TODO(exterkamp) add explanatory comment.
+   */
+  _updateTimingsForLightrider() {
+    // Bail if we aren't in Lightrider.
+    if (!global.isLightrider) return;
+
+    // Bail if timing is not initialized.
```

Comment:
include why?

Reply:
We can't process these numbers without timing, which should be initialized, but if it isn't we can't proceed, we could initialize it ourselves, but since it is already there I didn't think we'd need to, and just bail for safety if not.

Reply:
bump on this @exterkamp

Reply:
That's valid.  If there isn't timing info we can init to all -1.  I think this will be valid when we start to modify the timing data in follow up PR.  I'll remove this check for now since it isn't needed in this PR.